packages/playground/storage/src/lib/git-sparse-checkout.ts

```diff
@@ -260,10 +350,23 @@ export async function listGitRefs(
 			'content-type': 'application/x-git-upload-pack-request',
 			'Content-Length': `${packbuffer.length}`,
 			'Git-Protocol': 'version=2',
+			...getGitHubAuthHeaders(repoUrl),
```

Comment:
I'd rather not use any GitHub-specific concepts in this function. We could support injecting additional headers via arguments to keep it nice and generic